ipn/ipnlocal/local.go

```diff
@@ -3270,6 +3271,11 @@ func (b *LocalBackend) blockEngineUpdates(block bool) {
 // b.mu must be held.
 func (b *LocalBackend) reconfigAppConnectorLocked(nm *netmap.NetworkMap, prefs ipn.PrefsView) {
 	const appConnectorCapName = "tailscale.com/app-connectors"
+	defer func() {
+		if b.hostinfo != nil {
+			b.hostinfo.AppConnector.Set(b.appConnector != nil)
```

Comment:
the hostinfo handling in LocalBackend changed a fair bit recently to be slightly less dumpster fire-y than it historically has been but it's still pretty treacherous.

Note that Start blows this all away. Not sure whether you're restoring it all from that path or not.

Reply:
yeah, there's a set in start following the same pattern too